tokio/src/sync/tests/loom_watch.rs

```diff
@@ -34,3 +35,29 @@ fn smoke() {
         th.join().unwrap();
     })
 }
+
+#[test]
+fn wait_for_test() {
+    loom::model(move || {
+        let (tx, mut rx) = watch::channel(false);
+
+        let tx_arc = Arc::new(tx);
+        let tx1 = tx_arc.clone();
+        let tx2 = tx_arc.clone();
+
+        let th1 = thread::spawn(move || {
+            for _ in 0..3 {
+                tx1.send_modify(|_x| {});
```

Comment:
@Darksonn @debadree25 In this for loop its calling **send_modify** with an empty closure, which doesn't modify the value being sent. I think It would be more meaningful to actually modify the value being sent, so that the test exercises the wait_for function more thoroughly.  That makes sense, something like?:

 tx1.send_modify(|x| *x = true);
```suggestion
                tx1.send_modify(|x| *x = true);
```

Reply:
Not modifying the value is intentional, because it exercises the `wait_for` function more thoroughly. With your change, the `wait_for` method will _always_ succeed after the first call to `changed()`, whereas with the current version, `wait_for` may have to go around the loop several times.